services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -188,11 +166,122 @@ void testSaveBulkSubmission() throws IOException {
         bulkSubmissionService.saveBulkSubmission(bulkSubmission);
 
         // Verify the Mock
-        verify(mockGenericDao, times(1)).persist(bulkSubmission);
+        verify(mockGenericDao).persist(bulkSubmission);
+
+        assertTrue(true, EXPECTED_TO_PASS);
+    }
+
+    @Test
+    void testSaveBulkSubmissionWithDebug() throws IOException {
+        SdtContext.getContext().setRawInXml(Utilities.getRawXml(PATH_TEST_RESOURCES, TEST_XML_VALID_2));
+        ITargetApplication targetApplication = new TargetApplication();
+
+        Logger logger = (Logger) LoggerFactory.getLogger(BulkSubmissionService.class);
+
+        // Set logging level to debug so that afterCompletion logging is captured
+        logger.setLevel(DEBUG);
+
+        // Create appender and add to logger
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
+
+        // Mock the serviceRequest fetch
+        final IServiceRequest serviceRequest = new ServiceRequest();
+        when(mockGenericDao.fetch(IServiceRequest.class, 1)).thenReturn(serviceRequest);
+
+        // Activate Mock Generic Dao
+        final IBulkSubmission bulkSubmission = this.createBulkSubmission();
+        when(mockBulkCustomerDao.getBulkCustomerBySdtId(10)).thenReturn(bulkSubmission.getBulkCustomer());
+
+        final String key =
+            bulkSubmission.getBulkCustomer().getSdtCustomerId() + bulkSubmission.getCustomerReference();
+
+        // Setup concurrency map as if validator had done it.
+        IInFlightMessage inFlightMessage = new InFlightMessage();
+        inFlightMessage.setCompetingThreads(new HashMap<>());
+        inFlightMessage.getCompetingThreads().put(Thread.currentThread(), Thread.currentThread());
+
+        when(mockConcurrencyMap.get(key)).thenReturn(inFlightMessage);
+
+        // Put a dummy value into the SdtContext
+        SdtContext.getContext().setServiceRequestId(1L);
+
+        when(mockTargetApplicationDao.getTargetApplicationByCode(anyString())).thenReturn(targetApplication);
+
+        // Call the bulk submission service
+        bulkSubmissionService.saveBulkSubmission(bulkSubmission);
+
+
+        List<ILoggingEvent> logList = listAppender.list;
+
+        assertNotNull(logList.get(0).getFormattedMessage());
+
+        logger.detachAndStopAllAppenders();
+        // Verify the Mock
+        verify(mockGenericDao).persist(bulkSubmission);
+
+        assertTrue(true, EXPECTED_TO_PASS);
+    }
+
+
+
+   @Test
+    void testSaveBulkSubmissionTargetFound() throws IOException {
```

Comment:
Test name doesn't appear to match the purpose of the test.  This is very similar to the previous test except that it sets up some individual requests - possibly so that the for loops in enrich() and enqueueValidRequests() are not skipped.  These loops will be encountered whether the target is found or not, so the name doesn't quite match what it's doing.

Reply:
This was covering a branch on the code coverage so have renamed the Test method.